### PR TITLE
(docs) Overview of the language-tools

### PR DESCRIPTION
**Pull request by dummdidumm**
and how things work together

@orta @shirakaba @jasonlyu123 would be great if some of you could get me some feedback if everything is reasonable and understandable 😄

### CODE REVIEW

**Comment by orta** [comment]
This is an exceptionally well written document. Really well done.

File: docs/README.md [hunk 1]
```diff
@@ -96,3 +96,4 @@ Your default formatter for Svelte files may be wrong. Either it's set to the old
 ## Internals
 
 -   [Notes about deployment](./internal/deployment.md)
+-   [Overview of the language-tools and how things work together](./overview.md)
```
Comment (shirakaba) [inline docs/README.md:99]: Broken link; should be `./internal/overview.md` :)

File: docs/internal/overview.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+## language-server overview
+
+As briefly touched already, this is a language-server adhering to the [Language Server Protocol (LSP)](https://microsoft.github.io/language-server-protocol).
+The protocol defines the communication between an editor or IDE and a language server that provides language features like auto complete, go to definition, find all references etc.
+
+Our `language-server` can roughly be splitted into four areas:
+
+-   CSS: Provides intellisense for the things inside `<style>`. Internally it mostly forwards stuff to the `vscode-css-languageservice`.
+-   HTML: Provides intellisense for basic HTML tags like `div`, `a` etc. Internally it mostly forwards stuff to the `vscode-html-languageservice`.
```
Comment (shirakaba) [inline docs/internal/overview.md:26]: I’ve read through the rest and I’m not sure what the relation is between the `svelte2tsx` step and this `vscode-html-languageservice` stuff. I imagine they should overlap in some way.

File: docs/internal/overview.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+#### Integration of svelte2tsx into the language-server
+
+This example shows how our `language-server` uses `svelte2tsx`:
+
+Svelte file changes:
+
+1. Svelte file comes in
+2. Transform Svelte to TSX/JSX
+3. Pass that result to the TypeScript language service
```
Comment (shirakaba) [inline docs/internal/overview.md:61]: Would be nice to see a sample output of the TSX to get a picture of what magic is going on behind the scenes (e.g. generation of interfaces with reserved names) to make the Language Server understand Svelte. Would particularly help me understand how to contribute to the discussion on how to expose props and events typings from Svelte components.

File: docs/internal/overview.md (hunk 2 above)
Comment (dummdidumm) [reply docs/internal/overview.md]: They don't, I tried to clarify that.

File: docs/internal/overview.md (hunk 3 above)
Comment (dummdidumm) [reply docs/internal/overview.md:90]: Added a sample output and added some short mention of the ambient types. More details would go into a separate document purely for `svelte2tsx` - some day 😄

**Pull request merged**